Tb 3149 adapt settings widgets design

pro100svitlo opened the pull request:
### What 🕵️ 🔍

- This PR contains some changes to the xayn design, mainly related to the settings screen
- new colors, styles were added. <b>Partially</b>. Properly they will be added, renamed and sorted later, with the separate task.
- new button (ghost) was added (with docs)
- update settings related widgets
  - some components were temporarily deprecated. I left them in the codebase, cos later they will be added again.
  - changed style and colors
- update settings related example page

----------

### How to test 🥼 🔬

- compare design from figma

----------

### Screenshots 📸 📱

| Before | After  |
| ------ | ------ |
| <img width="280" src="https://user-images.githubusercontent.com/12133914/148563932-45b47a6a-c6d9-4299-a22a-40797b20b10f.png"> | <img width="280" src="https://user-images.githubusercontent.com/12133914/148563175-2265bf3a-835a-4bc1-bda7-bf939f9cfda2.png"> |
| <img width="280" src="https://user-images.githubusercontent.com/12133914/148563976-f6033c1b-1a55-4b1e-8708-8c6e62530ae9.png"> | <img width="280" src="https://user-images.githubusercontent.com/12133914/148563159-3e81ec93-a12a-472e-a781-8c61d7e5904f.png"> |
| <img width="280" src=""> | <img width="280" src="https://user-images.githubusercontent.com/12133914/148564301-342d66ab-eea7-4691-97d7-dc0d10eb9841.png"> |

----------
### References 📝 🔗

- [new colors scheme](https://www.figma.com/file/V9uNU8Bw8pKkkO9tmKvH3o/Design-System-%7C-Xayn?node-id=2008%3A1129)
- [figma settings](https://www.figma.com/file/hGAst3K9ZQIHmQoCNjbgaZ/Discovery-App-%7C-Design?node-id=391%3A1139)
- [jira task](https://xainag.atlassian.net/jira/software/projects/TB/boards/52?selectedIssue=TB-3149)

----------

pro100svitlo committed:
Add new resources: colors, styles. Update theme

pro100svitlo committed:
Add ghost button and example for it

pro100svitlo committed:
Update settings related widgets, tests and example page

joecks commented on example/lib/screen/xayn_widgets/page/settings_page.dart:
```diff
@@ -58,19 +61,19 @@ class _PrivacySectionState extends State<_PrivacySection> {
         ],
       );
 
+  // ignore: unused_element
   SettingsCardData _getBlockTrackers() {
     final tile = SettingsTileData(
```
❓  whats about those ignored elements?

pro100svitlo replied:
I decided to leave those elements in the codebase. 
From the talk with Adriel, I got that right now they are busy and do not have time for implementing all components, but soon they will renew them as well. So I decided to leave them and later just uncomment and modify them.

joecks commented on lib/src/linden/xcolors.dart:
```diff
@@ -160,4 +160,40 @@ class XColors {
 
   Brightness get brightness => _brightness;
+
+  /// right now we are in the middle of the design migration.
+  /// So far I added here some items, which I need for my tasks.
+  /// But later we will restructure all this file with proper items (names and structure)
+  //region NEW COLORS
```
💅  You could extend this class i.e. `NewColors extends XColors` and then pass a boolean to `Linden(newColors:true)` that would switch between the implementations, so you don't need to duplicate code.

pro100svitlo replied:
It is not just about switching between old and new. 
Right now UI team do not have all the color mapping scheme between the old and new palette. So its more like they add something new on top.

pro100svitlo replied:
But yes, I can try to do this as well 😉

joecks commented on lib/src/widget/settings/data/settings_tile_action.dart:
```diff
@@ -64,6 +66,7 @@ class SettingsTileActionCircle extends SettingsTileAction {
       ];
 }
 
+@Deprecated('So far there is no useCases, but later it might come back')
```
💅 Personally I would avoid colloquial language in such descriptions, i.e.: `Old Style Settings, might be removed`

pro100svitlo replied:
updated

joecks commented on lib/src/widget/settings/widget/settings_section.dart:
```diff
@@ -3,11 +3,12 @@ import 'package:xayn_design/xayn_design.dart';
 
-/// Should be equal to [XSizes.unit4];
-const _defaultTopPadding = 32.0;
+/// Should be equal to [XSizes.unit3];
+const _defaultTopPadding = 24.0;
```
❓  why can't we use XSizes?

pro100svitlo replied:
it is not `const`

joecks reviewed (approved):
Looks good. I just added some smaller suggestions.

pro100svitlo committed:
Add all colors from new palette

pro100svitlo committed:
Create new colors class and extend it from XColors

- update Linden class: reduce number of XColors object created
- add factory method to the XColors, to support new scheme creation
- update example app

pro100svitlo committed:
Change settings deprecated message

pro100svitlo committed:
Add new svg assets
